diff --git a/x264dll.c b/x264dll.c
index 5ffd109..10e94d9 100644
--- a/x264dll.c
+++ b/x264dll.c
@@ -24,8 +24,24 @@
  *****************************************************************************/
 
 #include "common/common.h"
+#include "config.h"
 #include <windows.h>
 
+static ALWAYS_INLINE void x264_process_attach_print_version()
+{
+    x264_log( NULL, X264_LOG_INFO, "H.264/MPEG-4 AVC codec - Copyleft 2003-2010 - http://www.videolan.org/x264.html\n" );
+    x264_log( NULL, X264_LOG_INFO, "version: %s\n", X264_POINTVER );
+    x264_log( NULL, X264_LOG_INFO, "built date: %s  (%s)\n", __DATE__, __TIME__ );
+#ifdef __GNUC__
+    x264_log( NULL, X264_LOG_INFO, "compiler: gcc %s\n", __VERSION__ );
+#else
+    x264_log( NULL, X264_LOG_INFO, "compiler: none-gcc\n" );
+#endif
+#ifdef X264_CFLAGS
+    x264_log( NULL, X264_LOG_INFO, "cflags: %s\n", X264_CFLAGS );
+#endif
+}
+
 /* Callback for our DLL so we can initialize pthread */
 BOOL WINAPI DllMain( HANDLE hinstDLL, DWORD fdwReason, LPVOID lpvReserved )
 {
@@ -33,7 +49,10 @@ BOOL WINAPI DllMain( HANDLE hinstDLL, DWORD fdwReason, LPVOID lpvReserved )
     switch( fdwReason )
     {
         case DLL_PROCESS_ATTACH:
-            pthread_win32_process_attach_np();
+            /* 
+             * pthread_win32_process_attach_np();
+             */
+            x264_process_attach_print_version();
 
         case DLL_THREAD_ATTACH:
             pthread_win32_thread_attach_np();
@@ -44,8 +63,10 @@ BOOL WINAPI DllMain( HANDLE hinstDLL, DWORD fdwReason, LPVOID lpvReserved )
             break;
 
         case DLL_PROCESS_DETACH:
-            pthread_win32_thread_detach_np();
-            pthread_win32_process_detach_np();
+            /*
+             * pthread_win32_thread_detach_np();
+             * pthread_win32_process_detach_np();
+             */
             break;
     }
 #endif
