diff --git a/Makefile b/Makefile
index bab55e5..1e2caaa 100644
--- a/Makefile
+++ b/Makefile
@@ -133,10 +133,10 @@ libx264.a: .depend $(OBJS) $(OBJASM)
 	$(RANLIB) libx264.a
 
 $(SONAME): .depend $(OBJS) $(OBJASM) $(OBJSO)
-	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
+	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(CFLAGS) $(LDFLAGS)
 
 x264$(EXE): $(OBJCLI) libx264.a
-	$(CC) -o $@ $+ $(LDFLAGSCLI) $(LDFLAGS)
+	$(CC) -o $@ $+ $(LDFLAGSCLI) $(CFLAGS) $(LDFLAGS)
 
 checkasm: tools/checkasm.o libx264.a
 	$(CC) -o $@ $+ $(LDFLAGS)
