diff --git a/common/common.c b/common/common.c
index 728dfab..b117781 100644
--- a/common/common.c
+++ b/common/common.c
@@ -161,6 +161,7 @@ void x264_param_default( x264_param_t *param )
     param->b_tff = 1;
     param->b_pic_struct = 0;
     param->b_fake_interlaced = 0;
+    param->i_open_gop = X264_OPEN_GOP_NORMAL;
 }
 
 static int x264_param_apply_preset( x264_param_t *param, const char *preset )
