*** libavcodec/h264.c.old	Sat Dec 19 22:15:48 2009
--- libavcodec/h264.c	Sat Dec 19 22:15:48 2009
***************
*** 8158,8163 ****
--- 8158,8174 ----
      return 0;
  }
  
+ /* MEANX */
+ int av_getAVCStreamInfo(AVCodecContext *avctx, uint32_t  *nalSize, uint32_t *isAvc)
+ {
+       H264Context *h = avctx->priv_data;
+       assert(h);
+       *nalSize=h->nal_length_size;
+       *isAvc=h->is_avc;
+ 	return 1;
+ 
+   }
+ /* MeanX */
  
  AVCodec h264_decoder = {
      "h264",
