diff --git a/Makefile b/Makefile
index 8074ce5..1684017 100644
--- a/Makefile
+++ b/Makefile
@@ -173,7 +173,7 @@ else
 fprofiled:
 	$(MAKE) clean
 	mv config.mak config.mak2
-	sed -e 's/CFLAGS.*/& -fprofile-generate/; s/LDFLAGS.*/& -fprofile-generate/' config.mak2 > config.mak
+	sed -e 's/CFLAGS.*/& -fprofile-generate -DX264_FPROFILE_GENERATE/; s/LDFLAGS.*/& -fprofile-generate/' config.mak2 > config.mak
 	$(MAKE) x264$(EXE)
 	$(foreach V, $(VIDS), $(foreach I, 0 1 2 3 4 5 6 7, ./x264$(EXE) $(OPT$I) --threads 1 $(V) -o $(DEVNULL) ;))
 	rm -f $(SRC2:%.c=%.o)
diff --git a/encoder/encoder.c b/encoder/encoder.c
index ae0bae3..f2e7a6b 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -1041,7 +1041,12 @@ x264_t *x264_encoder_open( x264_param_t *param )
         x264_log( h, X264_LOG_ERROR, "Are you attempting to run an SSE4a-targeted build on a CPU that\n" );
         x264_log( h, X264_LOG_ERROR, "doesn't support it?\n" );
 #endif
-        goto fail;
+#ifdef X264_FPROFILE_GENERATE
+        volatile int b_x264_fprofile_generate = 1;
+#else
+        volatile int b_x264_fprofile_generate = 0;
+#endif
+        if(!b_x264_fprofile_generate) goto fail;
     }
 
     h->out.i_nal = 0;
