diff --git a/encoder/encoder.c b/encoder/encoder.c
index 350b4e2..4c7e326 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -903,6 +903,23 @@ x264_t *x264_encoder_open( x264_param_t *param )
     /* Create a copy of param */
     memcpy( &h->param, param, sizeof(x264_param_t) );
 
+    /* Enable SubeME=10 */
+    if( h->param.analyse.i_subpel_refine == 9 && h->param.analyse.i_me_method >= X264_ME_UMH && h->param.analyse.i_trellis >= 2 )
+    {
+        x264_log( h, X264_LOG_WARNING, "QPRD (SubME=10) has been triggered!\n" );
+        h->param.analyse.i_subpel_refine++;
+    }
+
+    /* Disable Scenecuts */
+    if( h->param.i_keyint_max > 999 )
+    {
+        x264_log( h, X264_LOG_WARNING, "Non-scenecut keyframes have been disabled!\n" );
+        h->param.i_keyint_max = X264_KEYINT_MAX_INFINITE;
+    }
+
+    /* Apply "fast first-pass" settings */
+    /* x264_param_apply_fastfirstpass( &h->param ); */
+
     if( param->param_free )
         param->param_free( param );
 
