diff --git a/common/common.c b/common/common.c
index 4617387..f3f8f3a 100644
--- a/common/common.c
+++ b/common/common.c
@@ -65,7 +65,7 @@ void    x264_param_default( x264_param_t *param )
     param->i_level_idc     = -1;
     param->i_slice_max_size = 0;
     param->i_slice_max_mbs = 0;
-    param->i_slice_count = 0;
+    param->i_slice_count = 4;
 
     /* Encoder parameters */
     param->i_frame_reference = 3;
