diff --git a/common/common.c b/common/common.c
index 2d166af..7e08a79 100644
--- a/common/common.c
+++ b/common/common.c
@@ -158,7 +158,7 @@ void x264_param_default( x264_param_t *param )
     param->b_aud = 0;
     param->b_vfr_input = 1;
     param->b_dts_compress = 0;
-    param->i_nal_hrd = X264_NAL_HRD_NONE;
+    param->i_nal_hrd = X264_NAL_HRD_VBR;
     param->b_tff = 1;
     param->b_pic_struct = 0;
 }
