diff --git a/Makefile b/Makefile
index 5831091..1510be0 100644
--- a/Makefile
+++ b/Makefile
@@ -148,10 +148,10 @@ $(LIBX264): .depend $(OBJS) $(OBJASM)
 	$(if $(RANLIB), $(RANLIB) $@)
 
 $(SONAME): .depend $(OBJS) $(OBJASM) $(OBJSO)
-	$(LD)$@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
+	$(LD)$@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(CFLAGS) $(LDFLAGS)
 
 x264$(EXE): .depend $(OBJCLI) $(CLI_LIBX264)
-	$(LD)$@ $(OBJCLI) $(CLI_LIBX264) $(LDFLAGSCLI) $(LDFLAGS)
+	$(LD)$@ $(OBJCLI) $(CLI_LIBX264) $(CFLAGS) $(LDFLAGSCLI) $(LDFLAGS)
 
 checkasm: tools/checkasm.o $(LIBX264)
 	$(LD)$@ $+ $(LDFLAGS)
