*** libavformat/movenc.c.old	Sat Dec 19 22:16:15 2009
--- libavformat/movenc.c	Sat Dec 19 22:16:15 2009
***************
*** 2135,2137 ****
--- 2135,2149 ----
      .codec_tag = (const AVCodecTag* const []){codec_ipod_tags, 0},
  };
  #endif
+ /* MEANX */
+ int movenc_init(void)
+ {
+     av_register_output_format(&mov_muxer);
+     av_register_output_format(&tgp_muxer);
+     av_register_output_format(&mp4_muxer);
+     av_register_output_format(&psp_muxer);
+     av_register_output_format(&tg2_muxer);
+     av_register_output_format(&ipod_muxer);
+     return 0;
+ }
+ /* MEANX */
