diff --git a/encoder/encoder.c b/encoder/encoder.c
index 15b7e53..69ba5f0 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -881,6 +881,13 @@ x264_t *x264_encoder_open( x264_param_t *param )
         h->param.i_scenecut_threshold = 0;
     }
 
+    /* Force AutoVAQ */
+    if( h->param.rc.i_aq_mode )
+    {
+        x264_log( h, X264_LOG_WARNING, "AutoVAQ has been triggered!\n" );
+        h->param.rc.i_aq_mode = X264_AQ_AUTOVARIANCE;
+    }
+
     /* Apply "fast first-pass" settings */
     x264_param_apply_fastfirstpass( &h->param );
 
