diff --git a/x264.h b/x264.h
index 6609a32..774d088 100644
--- a/x264.h
+++ b/x264.h
@@ -35,7 +35,7 @@
 
 #include <stdarg.h>
 
-#define X264_BUILD 73
+#define X264_BUILD 72
 
 /* x264_t:
  *      opaque handler for encoder */
@@ -298,16 +298,16 @@ typedef struct x264_param_t
     int b_repeat_headers;       /* put SPS/PPS before each keyframe */
     int i_sps_id;               /* SPS and PPS id number */
 
-    /* Slicing parameters */
-    int i_slice_max_size;    /* Max size per slice in bytes; includes estimated NAL overhead. */
-    int i_slice_max_mbs;     /* Max number of MBs per slice; overrides i_slice_count. */
-    int i_slice_count;       /* Number of slices per frame: forces rectangular slices. */
-
     /* Optional callback for freeing this x264_param_t when it is done being used.
      * Only used when the x264_param_t sits in memory for an indefinite period of time,
      * i.e. when an x264_param_t is passed to x264_t in an x264_picture_t or in zones.
      * Not used when x264_encoder_reconfig is called directly. */
     void (*param_free)( void* );
+
+    /* Slicing parameters */
+    int i_slice_max_size;    /* Max size per slice in bytes; includes estimated NAL overhead. */
+    int i_slice_max_mbs;     /* Max number of MBs per slice; overrides i_slice_count. */
+    int i_slice_count;       /* Number of slices per frame: forces rectangular slices. */
 } x264_param_t;
 
 typedef struct {

--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -746,8 +746,13 @@
     CHECKED_MALLOCZERO( h, sizeof(x264_t) );
 
     /* Create a copy of param */
-    memcpy( &h->param, param, sizeof(x264_param_t) );
+    memcpy( &h->param, param, sizeof(x264_param_t) - 3 * sizeof(int) );
 
+    /* Workaround for slices support */
+    h->param.i_slice_max_size = 0;
+    h->param.i_slice_max_mbs = 0;
+    h->param.i_slice_count = 0;
+
     //Fast First Pass: --trellis 0 --no-8x8dct --partitions none --me dia --ref 1 --subme MIN(2, previous subme)
     if( h->param.rc.b_stat_write && !h->param.rc.b_stat_read )
     {

--- a/common/common.c
+++ b/common/common.c
@@ -39,7 +39,7 @@
 void    x264_param_default( x264_param_t *param )
 {
     /* */
-    memset( param, 0, sizeof( x264_param_t ) );
+    memset( param, 0, sizeof( x264_param_t ) - 3 * sizeof( int ) );
 
     /* CPU autodetect */
     param->cpu = x264_cpu_detect();
@@ -62,9 +62,6 @@
     param->i_fps_num       = 25;
     param->i_fps_den       = 1;
     param->i_level_idc     = -1;
-    param->i_slice_max_size = 0;
-    param->i_slice_max_mbs = 0;
-    param->i_slice_count = 0;
 
     /* Encoder parameters */
     param->i_frame_reference = 3;
