diff --git a/configure b/configure
index b35d120..b0ef182 100755
--- a/configure
+++ b/configure
@@ -702,6 +702,8 @@ Libs: $pclibs
 Cflags: -I$includedir
 EOF
 
+echo "#define X264_CFLAGS \"$CFLAGS\"" >> config.h
+
 cat > conftest.log <<EOF
 Platform:   $ARCH
 System:     $SYS
diff --git a/encoder/encoder.c b/encoder/encoder.c
index 92c1eed..e30b6ff 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -31,6 +31,7 @@
 #include "ratecontrol.h"
 #include "macroblock.h"
 #include "me.h"
+#include "config.h"
 
 #ifdef HAVE_VISUALIZE
 #include "common/visualize.h"
@@ -1118,6 +1119,20 @@ x264_t *x264_encoder_open( x264_param_t *param )
         h->sps->i_profile_idc == PROFILE_HIGH ? "High" :
         "High 4:4:4 Predictive", h->sps->i_level_idc/10, h->sps->i_level_idc%10 );
 
+    x264_log( h, X264_LOG_INFO, "version: %s\n", X264_POINTVER );
+    x264_log( h, X264_LOG_INFO, "built date: %s  (%s)\n", __DATE__, __TIME__ );
+#ifdef __GNUC__
+    x264_log( h, X264_LOG_INFO, "compiler: gcc %s\n", __VERSION__ );
+#else
+    x264_log( h, X264_LOG_INFO, "compiler: none-gcc\n" );
+#endif
+#ifdef X264_CFLAGS
+    x264_log( h, X264_LOG_INFO, "cflags: %s\n", X264_CFLAGS );
+#endif
+    char *print_opts = x264_param2string( &h->param, 1 );
+    x264_log( h, X264_LOG_INFO, "options: %s\n", print_opts );
+    x264_free( print_opts );
+
     return h;
 fail:
     x264_free( h );
