diff --git a/encoder/encoder.c b/encoder/encoder.c
index 36e5917..ea038d3 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -918,6 +918,13 @@ x264_t *x264_encoder_open( x264_param_t *param )
         h->param.i_scenecut_threshold = 0;
     }
 
+    /* Force AutoVAQ */
+    if( h->param.rc.i_aq_mode )
+    {
+        x264_log( h, X264_LOG_WARNING, "AutoVAQ has been triggered!\n" );
+        h->param.rc.i_aq_mode = X264_AQ_AUTOVARIANCE;
+    }
+
     /* Apply "fast first-pass" settings */
     /* x264_param_apply_fastfirstpass( &h->param ); */
 
