diff --git a/encoder/encoder.c b/encoder/encoder.c
index 539f74f..b602409 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -921,6 +921,13 @@ x264_t *x264_encoder_open( x264_param_t *param )
         h->param.i_keyint_max = X264_KEYINT_MAX_INFINITE;
     }
 
+    /* Force AutoVAQ */
+    if( h->param.rc.i_aq_mode )
+    {
+        x264_log( h, X264_LOG_WARNING, "AutoVAQ has been triggered!\n" );
+        h->param.rc.i_aq_mode = X264_AQ_AUTOVARIANCE;
+    }
+
     /* Apply "fast first-pass" settings */
     /* x264_param_apply_fastfirstpass( &h->param ); */
 
