diff --git a/Makefile b/Makefile
index 5831091..43829ef 100644
--- a/Makefile
+++ b/Makefile
@@ -195,7 +195,7 @@ fprofiled:
 else
 fprofiled:
 	$(MAKE) clean
-	$(MAKE) x264$(EXE) CFLAGS="$(CFLAGS) $(PROF_GEN_CC)" LDFLAGS="$(LDFLAGS) $(PROF_GEN_LD)"
+	$(MAKE) x264$(EXE) CFLAGS="$(CFLAGS) $(PROF_GEN_CC) -DX264_FPROFILE_GENERATE" LDFLAGS="$(LDFLAGS) $(PROF_GEN_LD)"
 	$(foreach V, $(VIDS), $(foreach I, 0 1 2 3 4 5 6 7, ./x264$(EXE) $(OPT$I) --threads 1 $(V) -o $(DEVNULL) ;))
 	rm -f $(SRC2:%.c=%.o)
 	$(MAKE) CFLAGS="$(CFLAGS) $(PROF_USE_CC)" LDFLAGS="$(LDFLAGS) $(PROF_USE_LD)"
diff --git a/encoder/encoder.c b/encoder/encoder.c
index 85ad046..882f360 100644
--- a/encoder/encoder.c
+++ b/encoder/encoder.c
@@ -1155,7 +1155,12 @@ x264_t *x264_encoder_open( x264_param_t *param )
         x264_log( h, X264_LOG_ERROR, "Are you attempting to run an SSE4a-targeted build on a CPU that\n" );
         x264_log( h, X264_LOG_ERROR, "doesn't support it?\n" );
 #endif
-        goto fail;
+#ifdef X264_FPROFILE_GENERATE
+        volatile int b_x264_fprofile_generate = 1;
+#else
+        volatile int b_x264_fprofile_generate = 0;
+#endif
+        if(!b_x264_fprofile_generate) goto fail;
     }
 
     h->out.i_nal = 0;
